compose-boundry.js

```diff
@@ -0,0 +1,44 @@
+class ComposeBoundary {
+	constructor(list, value) {
+		this.list = list;
+		this.value = value;
+	}
+
+	range(min, max, value) {
+		let inRange = this.checkRange(min, max, value);
+		return inRange;
+	}
+
+	checkRange(min, max, value) {
+		if (value >= min && value <= max) {
+			return true;
+		}
+	}
+
+	getStatus(list, value) {

```

Comment:
It takes quite some time to understand the code structure and flow. This could also lead to problems in future maintenance. Please think of simplifying the code flow.